tez-runtime-library/src/main/java/org/apache/tez/http/async/netty/AsyncHttpConnection.java

```diff
@@ -102,6 +103,15 @@ private void initClient(HttpConnectionParams httpConnParams) throws IOException
               .build();
           DefaultAsyncHttpClientConfig config = builder.build();
           httpAsyncClient = new DefaultAsyncHttpClient(config);
+          TezRuntimeShutdownHandler.addShutdownTask(new Thread(() -> {
+            try {
+              if (httpAsyncClient != null) {
+                httpAsyncClient.close();
```

Comment:
Should httpAsyncClient be set to "null" explicitly as well? Without it, it may create issues during container reuse. i.e httpAsyncClient will not be null, but in closed state & system would try to use it leading to exceptions.

Reply:
thanks @rbalamohan, makes sense, fixed